OrbitGl/TracepointTrack.cpp

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "TracepointTrack.h"
+
+#include "App.h"
+
+TracepointTrack::TracepointTrack(TimeGraph* time_graph) : Track(time_graph) {
+  color_ = Color(0, 255, 0, 255);
+}
+
+void TracepointTrack::UpdatePrimitives(uint64_t min_tick, uint64_t max_tick, PickingMode) {
+  Batcher* batcher = &time_graph_->GetBatcher();
+  const TimeGraphLayout& layout = time_graph_->GetLayout();
+  float z = GlCanvas::kZValueEvent;
+  float track_height = layout.GetEventTrackHeight();
+
+  ScopeLock lock(GOrbitApp->GetCaptureData().GetTracepointEventBufferMutex());
+
+  const std::map<uint64_t, orbit_client_protos::TracepointEventInfo>& tracepoints =
+      GOrbitApp->GetCaptureData().GetTracepointsOfThread(thread_id_);
+
+  const Color kWhite(255, 255, 255, 255);
+
+  for (auto it = tracepoints.lower_bound(min_tick); it != tracepoints.end(); ++it) {
```

Comment:
Why not
`for (auto it = tracepoints.lower_bound(min_tick); it != tracepoints.upper_bound(max_tick); ++it) {`?

That would make the inner loop even denser.

Reply:
Thank you! I will update the code.